core/src/main/scala/org/apache/spark/sql/delta/files/TransactionalWrite.scala

```diff
@@ -234,6 +239,34 @@ trait TransactionalWrite extends DeltaLogging { self: OptimisticTransactionImpl
 
     val committer = getCommitter(outputPath)
 
+    val partitionColNames = partitionSchema.map(_.name).toSet
+    // schema should be normalized, therefore we can do an equality check
+    val statsDataSchema = output.filterNot(c => partitionColNames.contains(c.name))
```

Comment:
nit: these two variable above ... are needed only when stats collection is enabled... then why create them before